### ISSUE TITLE
Design documents are not restored

### ISSUE THREAD

**Issue by roswah** [cloudant/couchbackup#336]
## Bug Description
When I restore from a backup all documents are restored/added to the new database except the design document containing two views.

### 1. Steps to reproduce

I have created a script that:
1. Deletes the db "demo". This is the db I want to restore from the backup in step 3.
2. Creates a new empty db called "demo"
3. Creates a backup from db demo_master and restores it contents to the db "demo".

The script contains the following code:

curl -u username:pwd -X DELETE http://localhost:5984/demo 
curl -u username:pwd -X PUT http://localhost:5984/demo 
couchbackup --db demo_master | couchrestore --db demo

There are two documents in demo_master. One "regular" document and one design document containing two views.

When I run the script it seems to have been executed successfully and I get the following information in the cmd:

=============================================================
D:\CouchDBDemoRestore>curl -u Administrator:Dismember! -X DELETE http://localhost:5984/demo
{"ok":true}

D:\CouchDBDemoRestore>curl -u Administrator:Dismember! -X PUT http://localhost:5984/demo
{"ok":true}

D:\CouchDBDemoRestore>couchbackup --db demo_master   | couchrestore --db demo

Performing backup on http://localhost:5984/demo_master using configuration:
{
  "bufferSize": 500,
  "log": "C:\\Users\\ADMINI~1.SYM\\AppData\\Local\\Temp\\tmp-13268y6jnNo7PifhP",
  "mode": "full",
  "parallelism": 5,
  "requestTimeout": 120000,
  "resume": false
}

  couchbackup:backup Fetching all database changes... +0ms

Performing restore on http://localhost:5984/demo using configuration:
{
  "bufferSize": 500,
  "parallelism": 5,
  "requestTimeout": 120000
}

  couchbackup:backup Total batches received: 1 +61ms
  couchbackup:backup Written batch ID: 0 Total document revisions written: 2 Time: 0.044 +11ms
  couchbackup:backup Finished - Total document revisions written: 2 +1ms
  couchbackup:restore restored 2 +0ms
  couchbackup:restore finished { total: 2 } +32ms

**What do I expect to happen?**

 I expect that both documents, even the design document is added to the restored database "demo". 

**What did actually happen?**
There was only one document added to the db "demo". The one that is not a design document.

I even splitted the backup and restore into two commands that I ran directly in the cmd. 
The two commands where:

- couchbackup --db demo_master --log demo-master-backup.log > demo-master-backup.txt
- demo-master-backup.txt | couchrestore --db demo

It did not work either even though I could see in the demo-master-backup.txt that the design document was there to.

**This is the content from the demo-master-backup.txt:**
[{"_id":"_design/trees","_rev":"2-f4aba845a8b88b824e855b936e4397fa","views":{"trees-by-company_code":{"map":"function(doc) {\n    if (doc.company_code) emit(doc.company_code, doc.name);\n}"},"trees-by-company_code-and-name":{"map":"function(doc) {\n    if (doc.company_code) emit([doc.company_code, doc.name], null)\n}"}},"language":"javascript","_revisions":{"start":2,"ids":["f4aba845a8b88b824e855b936e4397fa","f1bd9c363ef339b4ec6e0b1d9113b9d9"]}},{"_id":"f90924745d8d60295af454c00200330d","_rev":"4-30a7262adb2373c929b3bfc5d2f08e87","name":"Katalog från master","is_list":false,"company_code":"TOTALENTR","id":5831731514466,"children":[{"name":"Med en underkat","is_list":false,"id":5641950590995,"children":[{"name":"artiklar","is_list":true,"id":8242995543997,"children":[{"article_no":"4622213","supplier_company_code":"LEV_149","supplier_company_id":149}]}]}],"_revisions":{"start":4,"ids":["30a7262adb2373c929b3bfc5d2f08e87","852db51308c7b4cb64651e90dca0e59f","7fa8fd67ed3511c9cfa02f70c3758c67","98c61da95354ca4f7b38cba2948be97f"]}}]

**And this is the content from the log file:**
:t batch0 [{"id":"_design/trees"},{"id":"f90924745d8d60295af454c00200330d"}]
:changes_complete 6-g1AAAAIzeJyd0UsOgjAQBuAGSNSlJ9Aj9AWUldxE-zKVICyUtd5Eb6I30ZtgsS5KYgi6mSbTzJd_MiUAYGpCBeaybqRRIi9ryUtTH46l_Qo4EIu2bQsTcrC3jYmgMSYpVWDWVEpvd5VWA-NiaatY9QSMIWFZNlbIO2HdEzRClGd8rLDphNNHCN4CUyJmWn2bGZCqyFZwto_FLt5GcUIJkyPzOOXqlFunRG6rJIYSib8y3Z328DLBRGGif8r0dIp_bYwISaGvFC_d3Ku1
:d batch0

**Environment details:**
I have the latest version of @cloudant/couchbackup@ installed v. 2.5.2
I also have the latest version of node installed: 12.18.2

**Comment by bessbd** [comment]
Hi @roswah ,

Thank you for this ticket!
Can you please provide a little more information on your environment?
How is the service (CouchDB) provisioned / deployed? Is it via docker (`couchdb:3` or some other image)?
Also, how is the user created?

I'm asking, because according to https://docs.couchdb.org/en/stable/intro/security.html#:~:text=CouchDB%20has%20the%20idea%20of,users%20are%20allowed%20to%20do only admin users are allowed to create design docs.

**Comment by roswah** [comment]
Hi,

CouchDB is installed directly on a Windows Server 2016 without a docker.
So the CouchDB-folder with it´s bin-folder and everything that belongs to it is directly under the D:-drive on the server.

The user I used in the curl-command to delete and then create the database again is the same user that is the admin user in CouchDB. So when I go to  http://localhost:5984/_utility I login with this user. I created this user when I went to  http://localhost:5984/_utility the first time after the database was installed. This user is also an administrator on the server where CouchDB is installed. 

Is there someway I can provide user-credentials to the restore commando if this requires the admin user to do it?
Since I provide the user information to the delete and put command.

FYI: In this ticket I changed the username and password to a fictive to not reveal to much information but I see now that I missed to change it where I copied the response from the cmd.
